mani_skill/envs/utils/randomization/enhanced_distractors.py

```
@@ -0,0 +1,464 @@
+        sphere = actors.build_sphere(
+            scene,
+            initial_pose=sapien.Pose(),
+            name=f"distractor_sphere_{random.randint(0, 9999)}",
+            radius=properties["radius"],
+            color=properties["color"]
+        )
+        
+        # Set pose immediately
+        pose = sapien.Pose(p=position)
+        sphere.set_pose(pose)
+        
+        # Apply texture if configured
+        if textures_directory and random.random() < texture_probability:
+            TextureManager.apply_texture_to_objects([sphere._objs[0]], textures_directory)
+        
+        # Create object data for tracking
+        sphere_data = {
+            "object": sphere,
+            "position": position,
+            "size": properties["obj_size"]
+        }
+        
+        return sphere_data
+
+
+class EnhancedDistractorManager:
+    """Main class for managing enhanced distractors in the scene."""
+    
+    @staticmethod
+    def create_enhanced_distractors(scene, manipulation_obj_pos, cfg):
+        """
+        Create enhanced distractor objects (cylinders and spheres) and place them on the table.
+        
+        Args:
+            scene: The SAPIEN scene
+            manipulation_obj_pos: Position of the manipulation object
+            cfg: Configuration for enhanced distractors
+        
+        Returns:
+            List of internal objects data
+        """
+        manipulation_obj_size = [0.05, 0.05, 0.05]  # assumed size manipulation object (for overlap check only)
+
+        # Create list to track objects we've placed
+        existing_objects = [
+            {"position": manipulation_obj_pos, "size": manipulation_obj_size, "id": "manipulation_object"}
+        ]
+        
+        # Create storage for objects
+        internal_objects = []
+        
+        # Place distractors with an alternating approach to ensure variety
+        max_objects_to_place = cfg.get("max_objects", 4)  # Configurable number of objects
```

fail here instead of using default. configs should be correct